### ISSUE TITLE
Playground: incorrect type causing false negatives

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#4493]
```ts
/* eslint "@typescript-eslint/prefer-readonly-parameter-types": "error" */

function foo(arg: string[]) {}
```
[playground link](https://typescript-eslint.io/play/#ts=4.5.4&sourceType=module&code=PQKgBApgzgNglgOwC5gEQAEkE8AO0DGATnDkgLTTzLA6EQBmEhZdAhgCYD2CMWZOrQqwC2EJEzLY8UVAC40TQp0KowIYACgN9AK4J8SONzD1OnABSCA5vKhJiCKwG0AugEowAbwC+GoA&rules=N4IgAgLgngDgpgZwMYCcCWMIFpEBs0B2EA9DCnAGZwpbkCGAJgPYG5RYx0p0C2cE1LNHgIQALhDUUTFCAC+QA&tsConfig=N4IgzgLgTglgxhEAuaBXApgXyA)

This should error because the array is mutable. But it doesn't.

Debugging a bit - for some reason the type of `arg` is `{}`.
Maybe there's a config issue somewhere?

cc @armano2

**Comment by bradzacher** [comment]
The issue is that for some reason the program that is created for the lint process does not include the TS lib defs, even though the monaco TS instance does.

For example [here is a playground](https://typescript-eslint.io/play/#ts=4.5.2&sourceType=module&code=PQKgUApgzgNglgOwC5gATtQIgAJIJ4AO0AxgE5wFIC008ywCA9lQGYyMCGSiA5lQaUYBbOFGiYAXKgDaAcgilBpWQF0ANGgw58RKGQrVaiJA2YioAVzEATfoPPipchUtUaMWXIRLlKNWMamVBYIVhC2AG4cpFCSMvKKjMoqYCDAYGAIEADuqAAK9qIQABTFAJSoALwAfKgA3qgAvmUA3KjAwKiAMuSoTKzsXLx2wkVQqBkQAB4ESUioLCHE3IwIqGQQXBAFI2Ll9ZqkEEgWpKtZudsOpRU19U2t7Z2AoOSoh8en4eONGeubl0XlNodbq9ZhsTjcBB8AQ7aATaazeaLZarKDHFgsACMezqmmIKzRqAIVV6OXyhV2e1uDWaQM6PT6ITCkWiUDA30gCNIcw4UDwCGISIFKNQaIsGIATDjNBxshw4HNzuTYdcqrUaQ9gS9ZfKkOF2fCZtyhUs4CtUGF-rsCFIrRAADwhADWTGyCGqFTqjUeIMZoRsVCiMQylopJSVdtV1PumueRN5NgynKNPL5ApNIo4MBg0o8+IQhIAJBwSb89ZHY6htVAoHAeFlrJp0PmiwAjUuHP5hwE+6u1+v6ptrAlzaQcNSoVsqEk6hXKhwAOizOekxYnhanlZeBATnyQjHnRSX2fKQ+BDOYTIDQbZHIycBYqGKZa23bKnu959BVHMzOGDjZbIAAs4BgCAnxfCsPx9C8f1EP8YQAsBn07cs31QAB+VAkFICxwKkFgszEStYN-ANENGZDIPQgAfGjsNwiASO-Mjwn-SjigAQmolV3zpX0zHg8iwzZMBrAgYgYGicCFmFM1Vg4YhiAgSgoAANUYOBrAAYQAIWfVspCpWoIk06xWjARTlNUjStL0lCNjQ2FmL6VjbAosRRIyYEAGJDh4eTUB4dhWyzIkw2w7wxn3VBsiSJ1okYEJrGwoDwIIKS8B4QRksnCweFE0BwFQEBUAAJRUw4xGQaK0uHIQMqOQLGEfDgFLTYggMEJgrFQRgiFIQYVjQNIwGMBRCOUw8xHtAAVdUzxAYqPFKgBBJAkEUtKxmIZdQuIJ0xhYJJUvAqrGBgCxMwQaxgBOw4ACsJJFFrTumiAFyHErUGwHdBqEPqEAWGAWFAsCUtmurduzfanWwg8pgkq7wOAiBViQOq7VQURXmgC6InCT6PG+37og4AGVke57Pkh8DoZgWH4dQRHiGR2K0vRzGIpxqmlkJr7SuwN4TgLVBVve+YToxunhEal7H2AuBOuZgnSDWPbFLhnGWeR6wiZW9IPGlhA5oqywYCQTESVmidZrNiwLYlEkslV6pihWYHQezcIMKM4ookuiApFmm5ajt6AHct1B6LtAAZOAnQdcPzctj1o4tG6IFBht09CbMJ0piAnr56xfafYpOygFYpDavBQ9QZPI6dmOw3jxPTYjx20-o5Ks8QT56LzmAyltMMO5Tq36MbruWgyDwioFsWNq26BUBLenGeOtWVhgPA3t5+W3rtfWMEFv7ycBg+aahjWDqZnW9XZtGj+5sYr71xehaOEWxnFrGt7eviBqYFD7S3VjDTWJ90CjTzFwTq49I7OwgK7d2CB35l1KJXaua8EB1zVA3e2Ft05xwTknQhSBu4Z3EtnAevQHbD1HrCOa6dp4UNnt8capBJrgTtNpEcuElgnVxPPJai9xaHBYAoNGU0YpgJhPuHQH0BaGwwJ2awO897yMYIoxhDh7S12qLPRay1T6oG0qhVeJYIxhigSTc+AMH4nXFhvTWFobBM0QAqOAWY4AAC9wJyJsQ3ECO1b5azGDuGse44pr1IDwCwQg0ZICgBIReJZzowAJuAhmrjmRMwyVkwJsJYoKiAmvVAAc8J9TVmAqqiDXptW0WlNWHkIDuBWjglK6Si7PWyYzPJMUD5vVaSUjG5SYQRC0p8LBqwTouCSLYmBGAlRzTdo40gRkClByfJU7Zs1iGt1Ias+upktITgPpsjYVcEBl1ricsy9ytIj3eqsoxIiTHQLMRYsYJYsYYy4NjN+eNMmfGyKUnBsTBp71enUi2YxUYKWzH1R8RTJniRSnaNkxNSpbILmrd+T8FK4IljzHpJdFk-XsRUrMeFf4KUUBwaFj5MUUq-u8UW4trGwkWSo9Ay5mFTD1DdIFHB1EIF3hnF0jA3TSGnPRWVbtdkpIbs8u09oGhUDURomQeRsarETngV6s0VBSFWnKBU4Q5rSDyCoWojRDFzwwMCMQ2z4CtgXNACUAAGTEABWBcFrBqtjAguPWySzydH5fNf2NLoBSAAJJ6iDWBZhLdYRtyTtUD0uiihWttW8p1ojsVfMcpYiW-y5ikqriClK90yV6hSginBTLhmCDRZ8TFsSzrAoJo2Ytdbi4NtZVSpVYt6VQuRe9KArLhanF-qSC4NjlFDkGspAVkwhXWBFWKiVzpXQIFlenBVMbA7KpDjmmaZrdSWtmtIUIQhWwKFtQ6iNooICurgO6z1Pr-WBo4MGj6YasVOs6KuzNJ7aUJqTf+lN81VVj32YKtGW73oZvtIgSRasACqtQsJYdQMHF97zF7mNLT8hduNB2fBGQA1FUyUozOHWTAGMzgndt5OaIpDhYq8ko9TD+xa2U-zFhRglx9l0eAPswyopIUEzNubg+DTD5oFvQAvYtpHNjkaVFslKrSZ3fznSJ3T4VuUSdUT2koSm9GnOsERwtHzvqab1NpskJmaNSzqhM+j1LA5Mf+r5qp4t9Of1nRy0z3HgKMDEHqpNCAwpok2KgIQcDtqtsYO2vTS7sW8txtWgmqyIN7IOemo5cGL2ZvYWJCSUlDjDgLHMO0FW+ENYEfuUgRjfLIf8oFYKjBQowAi0USKuggA&rules=N4XyA&tsConfig=N4XyA) I created for https://github.com/typescript-eslint/typescript-eslint/issues/4553#issuecomment-1040651758.

You'll notice that if you delete all of the definitions below the `#region` comment, all the rule reports will disappear.
In a nutshell - in order to create a repro I manually copied the relevant lib defs into the playground.

**Comment by lonyele** [comment]
Just in case, please tackle this problem if you want because I'm out... I tried tweaking some part where lint was created or sandboxInstance or tried setting up with tsvfs but failed to find the solution...  Fix can be easy.. or maybe not.  I feel like #4414 is also highly related. good luck

**Comment by armano2** [comment]
> Just in case, please tackle this problem if you want because I'm out... I tried tweaking some part where lint was created or sandboxInstance or tried setting up with tsvfs but failed to find the solution... Fix can be easy.. or maybe not. I feel like #4414 is also highly related. good luck

sadly no, #4414 is unrelated to this one, and has to be handled in different part of code, 

currently we have 2 additional issues:
1. we are missing some options for editor, eg. `downlevelIteration` (easy fix by just adding it to json)
2. compiler options are not passed / updated in typescript program used for eslint (this is a little harder to fix)
